File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    const stack = [];
    let result = true;
    
    for(let value of s.split('')){
        if(value==='('){
            stack.push(value);
        }else if(value===')'){
            if(stack[stack.length-1]==='(')
                stack.pop();
            else
                result = false;
        }
    }
    
    if(stack.length>0) result = false
    
    return result;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    strings = list(s)
    stack = []
    
    for value in strings:
        if value == '(':
            stack.append(value)
        elif value == ')':
            if stack and stack[-1] == '(':
                stack.pop()
            else:
                return False
        else:
            return False    
        
    return len(stack) == 0

